prototype1/src/app_net/helpers.rs

```diff
@@ -11,7 +11,9 @@ mod helpers {
       .and_then(|external_ip| TcpStream::connect((external_ip, 80)))
       .and_then(|stream| {
         let addr = stream.local_addr();
-        stream.shutdown(Shutdown::Both);
+
+        // We don't care about the tcp connection, just the IP
+        let _ = stream.shutdown(Shutdown::Both);
```

Comment:
https://www.reddit.com/r/rust/comments/1zdabb/what_should_i_do_with_unused_result_which_must_be/ 

seems to be a good discussion on the matter. I'm really quite fond of the semantics of `let _`, because it tells the user "I don't care about the error, but please dont end my life if there is one." The other major approach is to unwrap, but the semantics of that are "Please destroy the universe if this thing fails, but I don't really think it'll actually happen". The final option is to handle both as first-class entities.

I wonder if I might be able to make you come around to `let _`.